File: BookSeeker_Backend/models/book.js
// Mapping Book Table
module.exports = (sequelize, DataTypes) => (
    sequelize.define('book', {
        bsin: {
            type: DataTypes.STRING(100),
            allowNull: false,
            unique: true,
        },
        title: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        author: {
            type: DataTypes.STRING(100),
            allowNull: false,
        },
        publisher: {
            type: DataTypes.STRING(100),
            allowNull: false,
        },
        introduction: {
            type: DataTypes.TEXT,
            allowNull: false,
        },
        cover: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        link: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        keyword: {
            type: DataTypes.TEXT,
            allowNull: false,
        },
        adult: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        genre: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        publication_date: {
            type: DataTypes.STRING,
            allowNull: false,
        }
    }, {
        timestamps: true, // 생성&수정일 기록
        paranoid: true, // 삭제일 기록
    })
);

File: BookSeeker_Front/app/src/main/java/com/example/bookseeker/model/data/Book.kt
package com.example.bookseeker.model.data

import android.os.Parcelable
import com.example.bookseeker.adapter.AdapterConstants
import com.example.bookseeker.adapter.ViewType
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.io.Serializable


// 도서 목록 데이터 생성 객체
@Parcelize
data class BookList(
    var page: Int,
    val results: ArrayList<BookData>) : Parcelable

// 도서 데이터 생성 객체
@Parcelize
data class BookData(
    val bsin: String,
    val title: String,
    val author: String,
    val publisher: String,
    val introduction: String,
    val cover: String,
    val link: String,
    val keyword: String,
    val adult: String,
    val genre: String,
    val publication_date: String,
    var rating: Float,
    var state: Int
) : ViewType, Parcelable {
    override fun getViewType() = AdapterConstants.BOOKS
}

// 도서 검색 데이터 생성 객체
data class BooksSearch(
    @SerializedName("keyword") val keyword: String
) : Serializable